File: 2nd Year/2nd Semester/MPP/PROIECT-C#/Lab1/Lab1/Service/ProbaService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Lab1.Domain;
using Lab1.Repository;

namespace Lab1.Service
{
    public class ProbaService
    {
        private readonly ProbaRepository probaRepository;

        public ProbaService(ProbaRepository probaRepository)
        {
            this.probaRepository = probaRepository;
        }

        public List<Proba> GetAllProbe()
        {
            return probaRepository.FindAll().ToList();
        }

        public TipProba? GetProbaForArbitru(Arbitru arbitru)
        {
            foreach (var proba in GetAllProbe())
            {
                if (proba.Arbitru.Id == arbitru.Id)
                {
                    return proba.TipProba;
                }
            }
            return null;
        }

        public Proba GetProbaByArbitru(Arbitru arbitru)
        {
            foreach (var proba in GetAllProbe())
            {
                if (proba.Arbitru.Id == arbitru.Id)
                {
                    return proba;
                }
            }
            return null;
        }
    }
}

File: 2nd Year/2nd Semester/MPP/Triatlon-Java/src/main/java/com/example/laborator/Service/ProbaService.java
package com.example.laborator.Service;

import com.example.laborator.Domain.Arbitru;
import com.example.laborator.Domain.Proba;
import com.example.laborator.Domain.TipProba;
import com.example.laborator.Repository.IProbaRepository;
import com.example.laborator.Utils.Observable;

import java.util.ArrayList;
import java.util.List;

public class ProbaService extends Observable {
    private final IProbaRepository probaRepository;

    public ProbaService(IProbaRepository probaRepository) {
        this.probaRepository = probaRepository;
    }

    public List<Proba> getAllProbe() {
        List<Proba> probe = new ArrayList<>();
        probaRepository.findAll().forEach(probe::add);
        return probe;
    }

    public TipProba getProbaForArbitru(Arbitru arbitru) {
        for (Proba proba : getAllProbe()) {
            if (proba.getArbitru().getId().equals(arbitru.getId())) {
                System.out.println("Proba gasita: " + proba.getTipProba());
                return proba.getTipProba();
            }
        }
        return null;
    }
}